### PR TITLE
Add basic "scipy" package support in ImplicitImports plugin

### PR DESCRIPTION
**Pull request by JorjMcKie**
Thank your for contributing to Nuitka!

!! Please check that you select the **develop branch** (details see below link) !!

Before submitting a PR, please review the guidelines:
https://github.com/kayhayen/Nuitka/blob/master/CONTRIBUTING.md

### PR Checklist
- [x] Correct base branch selected? `develop` for new features and bug fixes too. If it's
      part of a hotfix, it will be moved to ``master`` during it.
- [ ] All tests still pass. Check the developer manual about ``Running the Tests``. There
      are Travis tests that cover the most important things however, and you are
      welcome to rely on those, but they might not cover enough.
- [ ] Ideally new features or fixed regressions ought to be covered via new tests.
- [ ] Ideally new or changed features have documentation updates.

### CODE REVIEW

**Comment by JorjMcKie** [comment]
Change in ``Recursion.py``, function ``checkPluginSinglePath``: now use the ``module_kind`` value in call to ``recurseTo`` instead of a fixed value of ``py``. This addresses issue #308.
Adding scipy support also adresses #308, by adding some new implicitely imported modules.

Please note that a full Scipy support will require a new plugin - rather similar to numpy-plugin.

**Comment by JorjMcKie** [comment]
@kayhayen - could you review my change to ``Recursion.py`` please? There always is a fat chance that I am missing something crucial ...

File: UserPlugin-Creation.rst [hunk 1]
```diff
@@ -78,3 +78,80 @@ Sample invocation line and output::
 
     $
 
+Nuitka Options Overview
```
Comment (kayhayen) [inline UserPlugin-Creation.rst]: Wow! This is an amazing by product :)

File: nuitka/plugins/standard/ImplicitImports.py [hunk 2]
```diff
@@ -259,6 +259,11 @@ def getImplicitImports(self, module):
             yield "apt_pkg", True
         elif full_name == "numpy.core":
             yield "numpy.core._dtype_ctypes", False
+        elif full_name == "scipy.special":
+            yield "scipy.special._ufuncs_cxx", False
```
Comment (kayhayen) [inline nuitka/plugins/standard/ImplicitImports.py]: The `False` is really good for things that are not always that way. We don't explain that anywhere, do you?

File: nuitka/importing/Recursion.py [hunk 3]
```diff
@@ -275,7 +275,7 @@ def checkPluginSinglePath(plugin_filename, module_package):
                 module_filename=plugin_filename,
                 module_relpath=module_relpath,
                 module_package=module_package,
-                module_kind="py",
+                module_kind=module_kind,  # "py",
```
Comment (kayhayen) [inline nuitka/importing/Recursion.py]: As a comment, `# py` sucks of course, but I take, that it forgot to forward the `module_kind`, good catch.

**Comment by kayhayen** [comment]
@JorjMcKie Fantastic job, you absolutely exceeded all realistic expectations I had about the PR. Very good that you added so much documentation too.

**Comment by JorjMcKie** [comment]
Thanks a lot for your feedback!

That lonesome # "py" comment was an eyecatcher for you only. Should be deleted of course.

**Pull request merged**